scripts/lib/CIME/XML/generic_xml.py

```diff
@@ -292,6 +292,14 @@ def get_child(self, name=None, attributes=None, root=None, err_msg=None):
 
     def get_optional_child(self, name=None, attributes=None, root=None, err_msg=None):
         children = self.get_children(root=root, name=name, attributes=attributes)
+        if len(children) > 1:
```

Comment:
I'm wondering why this code is needed. get_children should already be able to down-select based on attributes.

Reply:
I also realize I should add this to get_child and will update with that changed.
If attributes is None then get_children doesn't check to see if the node has any attributes - it just returns all matching nodes.